torch/testing/_internal/common_device_type.py

```diff
@@ -718,25 +696,91 @@ class OpDTypes(Enum):
 #     operator's gradient formula supports
 #   OpDTypes.unsupported_backward - the test is instantiated for all dtypes the
 #     operator's gradient formula doesn't support
-#   OpDTypes.none - the test is instantied without any dtype. The dtype
-#     arg will be set to _NO_DTYPES.
+#   OpDTypes.none - the test is instantied without any dtype. The test signature
+#     should not include a dtype kwarg in this case.
 #
 # These options allow tests to have considerable control over the dtypes
 #   they're instantiated for. Finally, the @dtypes decorator composes with the
 #   @ops decorator, and works the same as the "dtypes" argument to @ops.
 
-class ops(object):
+class ops(_TestParametrizer):
     def __init__(self, op_list, *, dtypes: OpDTypes = OpDTypes.basic,
                  allowed_dtypes: Optional[Sequence[torch.dtype]] = None):
         self.op_list = op_list
         self.opinfo_dtypes = dtypes
         self.allowed_dtypes = set(allowed_dtypes) if allowed_dtypes is not None else None
 
-    def __call__(self, fn):
-        fn.op_list = self.op_list
-        fn.allowed_dtypes = self.allowed_dtypes
-        fn.opinfo_dtypes = self.opinfo_dtypes
-        return fn
+    def _parametrize_test(self, test, generic_cls, device_cls):
+        """ Parameterizes the given test function across each op and its associated dtypes. """
+        for op in self.op_list:
+            # Acquires dtypes, using the op data if unspecified
+            dtypes = device_cls._get_dtypes(test)
+            if dtypes is None:
+                if self.opinfo_dtypes == OpDTypes.unsupported_backward:
```

Comment:
I think some of these are no longer used and can probably be eliminated now, but that can be a separate issue/pr